services/motion/builtin/move_request.go

```diff
@@ -895,6 +903,21 @@ func (ms *builtIn) createBaseMoveRequest(
 		obstaclePollingFreq = time.Duration(1000/motionCfg.obstaclePollingFreqHz) * time.Millisecond
 	}
 
+	// anonymous function to determine if we are at the requested goal at the start and end of plan's execution
+	atGoalCheck := func(basePose spatialmath.Pose) *state.ExecuteResponse {
+		if valExtra.motionProfile == motionplan.PositionOnlyMotionProfile {
```

Comment:
This is the reason that the handling of this should make its way into `motionplan` rather than here. We're relying on specific interpretations of `extra`, magic numbers, and hoping that nothing more is added in the motionplan that would invalidate this.

The PR as written is fine for fixing the flaky test, but we should put TODOs here due to how fragile this is.

Reply:
understood, I've added a todo right above this check so we know that it should be moved into the motionplan package.

Reply:
I also added another todo in the moveRequest struct re-iterating this point.